guillotina/contentnegotiation.py

```diff
@@ -1,844 +1,34 @@
-            for p in possibilities:
-                # for each accept parameter with the same q value
-
-                # find out if the possibility p matches anything in the server.  This uses
-                # the AcceptParameter's matches() method which will take into account
-                # wildcards, so content types like */* will match appropriately.  We ge
-                # back from this the concrete AcceptParameter as specified by the server
-                # if there is a match, so we know the result contains no unintentional wildcards
-                match = self._contains_match(p, server)
-                if match is not None:
-                    # if there is a match, register it
-                    allowable.append(match)
-
-            log.info("Allowable: " + str(q) + ":" + str(allowable))
-
-            # we now know if there are 0, 1 or many allowable content types at this q value
-            if len(allowable) == 0:
-                # we didn't find anything, so keep looking at the next q value
-                continue
-            elif len(allowable) == 1:
-                # we found exactly one match, so this is our content type to use
-                return allowable[0]
-            else:
-                # we found multiple supported content types at this q value, so now we need
-                # to choose the server's preference
-                for i in range(len(server)):
-                    # iterate through the server explicitly by numerical position
-                    if server[i] in allowable:
-                        # when we find our first content type in the allowable list, it is
-                        # the highest ranked server content type that is allowable, so
-                        # this is our type
-                        return server[i]
-
-        # we've got to here without returning anything, which means that the client and
-        # server can't come to an agreement on what content type they want and can deliver.
-        # There's nothing more we can do!
-        return None
-
-
-@implementer(IContentNegotiation)
-class ContentNegotiatorUtility(object):
-
-    def __init__(self, header, enabled_values):
-        if header == 'content_type':
-            server = [AcceptParameters(content_type=ContentType(x)) for x in enabled_values]
-            self.cn = ContentNegotiator(acceptable=server)
-        if header == 'language':
-            server = [AcceptParameters(language=Language(x)) for x in enabled_values]
-            self.cn = ContentNegotiator(acceptable=server)
-
-    def negotiate(self, **kwargs):
-        return self.cn.negotiate(**kwargs)
-
-
-def content_type_negotiation(request, resource, view):
+def get_ordered_content_type_value_list(value):
+    '''
+    ca-ES,ca;q=0.9,en;q=0.8,es;q=0.7
+    '''
+    result = []
+    for ct in value.lower().split(','):
+        weight = 1.0
+        ct_name = ct.split(';')[0]
+        for param in ct.split(';')[1:]:
```

Comment:
There is the use case were accept can be:     Accept: text/*, text/plain, text/plain;format=flowed, */*
[https://tools.ietf.org/html/rfc7231#section-5.3.2](https://tools.ietf.org/html/rfc7231#section-5.3.2)